github/src/main/kotlin/gropius/sync/github/Outgoing.kt

```diff
@@ -0,0 +1,225 @@
+import kotlinx.coroutines.reactor.awaitSingle
+import org.neo4j.cypherdsl.core.Cypher
+import org.slf4j.LoggerFactory
+import org.springframework.beans.factory.annotation.Qualifier
+import org.springframework.data.mongodb.core.ReactiveMongoOperations
+import org.springframework.data.neo4j.core.ReactiveNeo4jOperations
+import org.springframework.stereotype.Component
+
+/**
+ * Stateless component for the incoming part of the sync
+ * @param helper Reference for the spring instance of JsonHelper
+ * @param imsConfigManager Reference for the spring instance of IMSConfigManager
+ * @param neoOperations Reference for the spring instance of ReactiveNeo4jOperations
+ * @param syncNotificator Reference for the spring instance of SyncNotificator
+ * @param tokenManager Reference for the spring instance of TokenManager
+ * @param repositoryInfoRepository Reference for the spring instance of RepositoryInfoRepository
+ * @param issueInfoRepository Reference for the spring instance of IssueInfoRepository
+ * @param userInfoRepository Reference for the spring instance of UserInfoRepository
+ * @param labelInfoRepository Reference for the spring instance of LabelInfoRepository
+ * @param mongoOperations Reference for the spring instance of ReactiveMongoOperations
+ * @param timelineEventInfoRepository Reference for the spring instance of TimelineEventInfoRepository
+ * @param issueCleaner Reference for the spring instance of IssueCleaner
+ * @param nodeSourcerer Reference for the spring instance of NodeSourcerer
+ * @param timelineItemHandler Reference for the spring instance of TimelineItemHandler
+ */
+@Component
+class Outgoing(
+    private val repositoryInfoRepository: RepositoryInfoRepository,
+    private val issueInfoRepository: IssueInfoRepository,
+    private val userInfoRepository: UserInfoRepository,
+    private val labelInfoRepository: LabelInfoRepository,
+    private val mongoOperations: ReactiveMongoOperations,
+    private val timelineEventInfoRepository: TimelineEventInfoRepository,
+    private val issueCleaner: IssueCleaner,
+    private val nodeSourcerer: NodeSourcerer,
+    private val timelineItemHandler: TimelineItemHandler,
+    @Qualifier("graphglueNeo4jOperations")
+    private val neoOperations: ReactiveNeo4jOperations,
+    private val helper: JsonHelper,
+    private val imsConfigManager: IMSConfigManager,
+    private val syncNotificator: SyncNotificator,
+    private val tokenManager: TokenManager,
+    private val issueRepository: IssueRepository,
+    private val timelineItemRepository: TimelineItemRepository,
+    private val imsUserRepository: IMSUserRepository,
+    private val nodeDefinitionCollection: NodeDefinitionCollection,
+    private val incoming: Incoming
+) {
+    /**
+     * Logger used to print notifications
+     */
+    private val logger = LoggerFactory.getLogger(Outgoing::class.java)
+
+    private fun findFinalTypeBlock(relevantTimeline: List<TimelineItem>): List<TimelineItem> {
```

Comment:
docs

Reply:
actually for the whole file

Reply:
b77ec650cf2ee7b86c3a6b5010b3563aae0d4070